resources/js/Components/Form/CampaignCreateForm.tsx

```diff
@@ -0,0 +1,141 @@
+import {useForm} from '@inertiajs/inertia-react'
+import {Button, Grid, TextField, Typography} from '@mui/material'
+import {Autocomplete, ErrorText, Select} from 'Components'
+import React from 'react'
+import styled from 'styled-components'
+import {adventureType} from 'Types/adventure-data'
+import {CampaignData} from 'Types/campaign-data'
+import {CharacterData} from 'Types/character-data'
+import route from 'ziggy-js'
+
+type CampaignCreateFormPropType = {
+    type: 'Edit' | 'Create'
+    onCloseDrawer?: () => void
+    editData?: CampaignData
+    editId?: number
+    adventures: adventureType[]
+    characters: CharacterData[]
+}
+
+type CampaignFormDataType = {
+    title: string | undefined
+    character_id: number | null
+    adventure: adventureType | undefined
+    [key: string]: any
+}
+
+const StyledGrid = styled(Grid)`
+    margin-bottom: 16px;
+`
+
+const CampaignCreateForm = ({
+    type,
+    onCloseDrawer,
+    editData,
+    editId = 0,
+    adventures,
+    characters,
+}: CampaignCreateFormPropType) => {
+    const CAMPAIGN_CREATE_FORM_INITIAL_VALUE: CampaignFormDataType = {
+        title: undefined,
+        character_id: null,
+        adventure: undefined,
+    }
+    const CAMPAIGN_INITIAL_VALUE: CampaignFormDataType =
+        type === 'Create'
+            ? CAMPAIGN_CREATE_FORM_INITIAL_VALUE
+            : {
+                  title: editData?.title,
+                  character_id: editData?.characters[0].id || null,
```

Comment:
hmm, im unsure about this, we should probably use a select that accepts multiple values for this

Reply:
because backend accepts multiple characters in one campaign i believe